### ISSUE TITLE
AttributeError in dox2html5.py

### ISSUE THREAD

**Issue by codecat** [mosra/m.css#96]
The source I'm trying to parse here is a bit larger than my previous source which worked fine. Unsure what in the source could be causing this.

* Latest master branch (`bf8dfed`)
* Python 3.6.8
* Doxygen 1.8.15

If desired, I can upload (or email?) the (giant) auto-generated header file that causes this exception.

Below is the final output of the script:

```
Generating XML output for file scripts_docs.h
lookup cache used 13070/65536 hits=57543 misses=14049
finished...
Traceback (most recent call last):
  File "/home/nimble/dev/m.css/doxygen/dox2html5.py", line 3668, in <module>
    run(doxyfile, os.path.abspath(args.templates), args.wildcard, args.index_pages, search_merge_subtrees=not args.search_no_subtree_merging, search_add_lookahead_barriers=not args.search_no_lookahead_barriers, search_merge_prefixes=not args.search_no_prefix_merging)
  File "/home/nimble/dev/m.css/doxygen/dox2html5.py", line 3552, in run
    parsed = parse_xml(state, file)
  File "/home/nimble/dev/m.css/doxygen/dox2html5.py", line 2678, in parse_xml
    class_.name = symbol.leaf_name if state.compounds[compound.id].parent and symbol.parent.startswith(state.compounds[compound.id].parent) else symbol.name
AttributeError: 'NoneType' object has no attribute 'startswith'
```

**Comment by mosra** [comment]
Haha oh well. Some internal state mismatch, it seems. Not sure what the problem could be without seeing the input.

Yes, can you upload both the header file and the generated XML? E-mail is fine also: mosra@centrum.cz

**Comment by codecat** [comment]
I've sent you a zip file through mail. Thanks!

**Comment by mosra** [comment]
You were right when you said *giant*, ha :)

![image](https://user-images.githubusercontent.com/344828/55254347-e6d71b00-5257-11e9-9ad5-95e2830a7a30.png)

Should be fixed in 07a027fac66e5ab77a31016336e89568a2248679 along with a regression test case. Problem was that a namespaced class had a base in the root namespace and the code was not checking enough. Took me a while to dig that out :D

As before, please complain if it's *still* not working for you ;)

**Comment by codecat** [comment]
Works! Thanks for the super quick fix!
